app/services/finance/ecf/mentor/statement_calculator.rb

```diff
@@ -0,0 +1,113 @@
+# frozen_string_literal: true
+
+module Finance
+  module ECF
+    module Mentor
+      class StatementCalculator
+        class << self
+          def event_types
+            %i[
+              started
+              completed
+            ]
+          end
+
+          def event_types_for_display
+            %i[
+              started
+              completed
+            ]
+          end
+        end
+
+        attr_reader :statement
+
+        def initialize(statement:)
+          @statement = statement
+        end
+
+        def vat
+          total * vat_rate
+        end
+
+        def voided_declarations
+          statement.participant_declarations.voided.where(type: "ParticipantDeclaration::Mentor")
+        end
+
+        event_types.each do |event_type|
+          define_method "#{event_type}_fee_for_declaration" do
+            fee_for_declaration(type: event_type)
+          end
+
+          define_method "additions_for_#{event_type}" do
```

Comment:
Would it be a big change to get rid of these dynamic method names? open to opinions, but I find they make searching/navigating the code much harder